addons/web/static/src/report/report_client_action.js

```diff
@@ -0,0 +1,88 @@
+/** @odoo-module alias=report.client_action **/
+
+import ControlPanel from "web.ControlPanel";
+
+import { registry } from "@web/core/registry";
+import { breadcrumbsToLegacy } from "@web/legacy/utils";
+import { useService } from "@web/core/service_hook";
+import * as reportUtils from "./utils";
+const { Component, tags, hooks } = owl;
+export class ReportAction extends Component {
+    setup() {
+        this.actionService = useService("action");
+        this.userService = useService("user");
+        this.title = this.props.options.display_name || this.props.options.name;
+        this.trusted_origin = reportUtils.buildOrigin(
+            reportUtils.getProtocolFromUrl(this.userService.web_base_url),
+            reportUtils.getHostFromUrl(this.userService.web_base_url)
+        );
+        this.report_url = this.props.options.report_url;
+    }
+
+    mounted() {
+        window.addEventListener("message", this.onMessageReceived.bind(this));
```

Comment:
useExternalListener in constructor ?
And should not we bind the handler on the iframe ? by using HTMLIFrameElement.contentWindow (to reference an embedded <iframe> from its parent window), from the clientAction and not the other way around ?